silence/silence_test.go

```diff
@@ -974,23 +981,40 @@ func TestSilenceExpireWithZeroRetention(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 1, count)
 
+	count, err = s.CountState(types.SilenceStateActive)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
+
 	count, err = s.CountState(types.SilenceStateExpired)
 	require.NoError(t, err)
 	require.Equal(t, 1, count)
 
+	// Advance time. The silence state management code uses update time when
+	// merging, and the logic is "first write wins". So we must advance the clock
+	// one tick for updates to take effect.
+	clock.Add(1 * time.Millisecond)
+
 	require.NoError(t, s.Expire("pending"))
 	require.NoError(t, s.Expire("active"))
-
 	require.NoError(t, s.Expire("expired"))
 
+	// Advance time again. Despite what the function name says, s.Expire() does
+	// not expire a silence. It sets the silence to EndAt the current time. This
+	// means that the silence is active immediately after calling Expire.
+	clock.Add(1 * time.Millisecond)
+
+	// Make sure we can still find our silences.
 	_, err = s.QueryOne(QIDs("pending"))
 	require.NoError(t, err)
```

Comment:
do we really need this check? I believe that counting the number of silenes per state is enough.

Reply:
I agree with you. If this check was exhaustive you'd do it three times, or include all three QIDs.